umh-core/pkg/config/manager.go

```diff
@@ -38,16 +38,41 @@ const (
 	DefaultConfigPath = "/data/config.yaml"
 )
 
+// singleton instance
+// we avoid, having more than one instance of the config manager because it can lead to race conditions
+// if we ensure, that we have only one instance, we can avoid race conditions by using mutexes in this single instance as we do here
+
+// however, access from outside the package is not protected by mutexes (keep in mind e.g. when using GitOps on the config file)
+var (
+	instance ConfigManager
+	once     sync.Once
+)
+
 // ConfigManager is the interface for config management
 type ConfigManager interface {
 	// GetConfig returns the current config
 	GetConfig(ctx context.Context, tick uint64) (FullConfig, error)
-	// WriteConfig writes the current config to the file
-	WriteConfig(ctx context.Context, config FullConfig) error
 	// AtomicSetLocation sets the location in the config atomically
 	AtomicSetLocation(ctx context.Context, location models.EditInstanceLocationModel) error
 }
 
+// GetConfigManager returns the singleton instance of ConfigManager
+func GetConfigManager() ConfigManager {
+	once.Do(func() {
+		// Default to FileConfigManagerWithBackoff as it's the more robust implementation
```

Comment:
produce error message if the instance already exists

Reply:
what about this here? this doesn't produce an error yet, so it can lead to incorrect implemenetations. This here should fail if GetConfigManager is called a second time.